src/game_renderer/GameRenderer.cpp

```diff
@@ -133,12 +135,17 @@ void GameRenderer::Render(std::shared_ptr<GameState> game_state) {
   glUniformMatrix4fv(current_program->getUniform("V"), 1, GL_FALSE,
                      glm::value_ptr(V.topMatrix()));
 
-  for (std::shared_ptr<Platform> platform : *level->getPlatforms()) {
-    if (!ViewFrustumCulling::IsCulled(platform->GetBoundingBox(), vfplane)) {
-      MV = platform->GetTransform();
-      glUniformMatrix4fv(current_program->getUniform("MV"), 1, GL_FALSE,
-                         glm::value_ptr(MV.topMatrix()));
-      platform->GetModel()->draw(current_program);
+  for (std::shared_ptr<GameObject> obj : *level->getObjects()) {
+    if (obj->isObstacle()) {
+      if (Platform* platform = dynamic_cast<Platform*>(obj.get())) {
```

Comment:
If we are going with dynamic_casting, then what's the point of having/using isObstacle()/GetType() here? The if() check around the dynamic cast should take care of it, right?